frontend/src/metabase-lib/queries/NativeQuery.ts

```diff
@@ -433,7 +433,7 @@ export default class NativeQuery extends AtomicQuery {
 
   updateSnippetsWithIds(snippets): NativeQuery {
     const tagsBySnippetName = _.chain(this.templateTags())
-      .filter(tag => tag.type === "snippet" && tag["snippet-id"] == null)
```

Comment:
From what I tested, it seems like this extra null check was preventing snippets from being updated with new IDs. for example, given a SQL query 

```
select * from {{snippet: Table: Orders}} limit 1
``

and snippets
```
Table: Orders --> ID: 1
Table: Reviews  --> ID: 2
```

The dataset query stored in state would refer to the snippet ID `1`.

If we were to manually replace the snippet (i.e. just deleting `Orders` and replacing it with `Reviews`):
```
select * from {{snippet: Table: Reviews}} limit 1
```
With the extra null check, the snippet would be updated with the new information about the snippet _but the snippet ID would still be `1`_. Removing this check updates the snippet ID to `2`.